src/domains.jl

```diff
@@ -31,19 +31,20 @@ Constructs a OneDGrid object with size `Lx`, resolution `nx`, and leftmost
 position `x0`. FFT plans are generated for `nthreads` CPUs using
 FFTW flag `effort`.
 """
-struct OneDGrid{T<:AbstractFloat, Ta<:AbstractArray, Tfft, Trfft} <: AbstractGrid{T, Ta}
+struct OneDGrid{T<:AbstractFloat, Tk, Tx, Tfft, Trfft} <: AbstractGrid{T, Tk}
```

Comment:
Where is the second type parameter in `AbstractGrid` used? What's its purpose?

Reply:
Some functions use this type of array (`Array` or `CuArray`), e.g.,
https://github.com/FourierFlows/FourierFlows.jl/blob/999b1ef28a3d1010d7bf9ae31c48a65fd708f321/src/domains.jl#L277-L293

Reply:
It's also used a lot in GeophysicalFlows.jl